### ISSUE TITLE
License icons/titles accessibility

### ISSUE THREAD

**Issue by obulat** [WordPress/openverse-frontend#358]
## Problem

We display licenses using icons and their names in several places, but they are not accessible. 

For example, if you open the search filter sidebar with VoiceOver in Firefox, you will hear "Circle C C Circle Human Figure Circle Dollar sign with overlay backslash Circle equal sign  B Y N C N D unchecked checklist" for  CC BY-NC-ND, with each letter pronounced separately. When reading the license label, VoiceOver pronounces CC BY-SA license name as "B Y S A text element", and it is very difficult to understand that it refers to a license.

## Description

We are using two similar components to display licenses (name and icons): `src/components/License/License.vue` and `src/components/LicenseIcons.vue`. `LicenseIcons` is using VocabularyIcons icon font which we have decided to deprecate, so the solution should be to:
- deprecate `LicenseIcons` component, and use License component everywhere instead.
- make the `License` component more flexible to allow it to be used both as a span and as a checkbox label
- add a screen-reader-friendly label to the licenses: instead of the short abbreviation, it should use the full license name (eg. "CC BY-SA" -> "CC Attribution Share-Alike license")
- make the actual svgs hidden from screen readers.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

**Comment by dhruvkb** [comment]
Just wanted to add that there are some changes to `License` in #399 so it might help with conflict-resolution to branch out from `global_audio` instead of `main`.
